### ISSUE TITLE
Param docblocks should still be inherited from interfaces/parents when PHP types are present

### ISSUE THREAD

**Issue by muglug** [vimeo/psalm#1213]
```php
class A {}
class B extends A {}
  
interface X {
    /**
     * @param B $class
     */
    public function boo(A $class): void;
}

class Y implements X {
    public function boo(A $class): void {}
}

(new Y())->boo(new A);
```

```php
class A {}
class B extends A {}
  
class X {
    /**
     * @param B $class
     */
    public function boo(A $class): void {}
}

class Y extends X {
    public function boo(A $class): void {}
}

(new Y())->boo(new A)
```

https://getpsalm.org/r/851ac86450

**Comment by muglug** [comment]
Actually I already decided against this in https://github.com/vimeo/psalm/issues/1100 cc @ro0NL

**Comment by ro0NL** [comment]
@muglug the issue is at least an explicit `@inheritdoc` / `{@inheritdoc}` is ignored, causing phpdoc duplication basically :cry: 

https://getpsalm.org/r/d23255ed7a

**Comment by muglug** [comment]
You're right, _that_ should be a bug

**Comment by muglug** [comment]
Fixed in 11dc78fd766ad5562660f49771e736806d2581ef and 4f3d7fca0d906d9fc1236e19ef7e9314d9909cae
